src/Uno.UWP/Storage/Pickers/FileOpenPicker.iOS.cs

```diff
@@ -75,6 +63,55 @@ private async Task<FilePickerSelectedFilesArray> PickFilesAsync(bool multiple, C
 			return new FilePickerSelectedFilesArray(files);
 		}
 
+		private UIViewController GetViewController(bool multiple, TaskCompletionSource<NSUrl?[]> completionSource)
+		{
+			switch (SuggestedStartLocation)
+			{
+				case PickerLocationId.PicturesLibrary:
+					return new UIImagePickerController()
+					{
+						SourceType = UIImagePickerControllerSourceType.PhotoLibrary,
+						MediaTypes = UIImagePickerController.AvailableMediaTypes(UIImagePickerControllerSourceType.PhotoLibrary),
+						ImagePickerControllerDelegate = new ImageOpenPickerDelegate(completionSource)
+					};
+
+				default:
+					var documentTypes = UTTypeMapper.GetDocumentTypes(FileTypeFilter);
+					return new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open)
+					{
+						AllowsMultipleSelection = multiple,
+						ShouldShowFileExtensions = true,
+						Delegate = new FileOpenPickerDelegate(completionSource)
+					};
+			}
+		}
+
+		private class ImageOpenPickerDelegate : UIImagePickerControllerDelegate
+		{
+			private readonly TaskCompletionSource<NSUrl?[]> _taskCompletionSource;
+
+			public ImageOpenPickerDelegate(TaskCompletionSource<NSUrl?[]> taskCompletionSource) =>
+				_taskCompletionSource = taskCompletionSource;
+
+			public override void Canceled(UIImagePickerController picker) =>
+				_taskCompletionSource.SetResult(Array.Empty<NSUrl?>());
```

Comment:
I see you're following the other delegate implementation, but it doesn't seem correct.

```suggestion
				_taskCompletionSource.SetCanceled();
```

Reply:
`SetCanceled` may require extra validation and more code somewhere else to properly detect for cancellation (awaiting a canceled task would throw `TaskCanceledException`). Let's keep as-is for now.

Reply:
It will require more work and can be argued that it brings no value. So let's keep both as-is.

Reply:
if I use SetCanceled(), PickMultipleFilesTaskAsync does not resolved.